recipes/thrift/all/conanfile.py

```diff
@@ -0,0 +1,145 @@
+        if self.options.with_libevent:
+            self.requires("libevent/2.1.11")
+
+    def _configure_cmake(self):
+        if self._cmake:
+            return self._cmake
+        self._cmake = CMake(self)
+        for option, value in self.options.items():
+            self._cmake.definitions[option.upper()] = value
+
+        # Make thrift use correct thread lib (see repo/build/cmake/config.h.in)
+        self._cmake.definitions["USE_STD_THREAD"] = self.options.with_stdthreads
+        self._cmake.definitions["USE_BOOST_THREAD"] = self.options.with_boostthreads
+        self._cmake.definitions["WITH_SHARED_LIB"] = self.options.shared
+        self._cmake.definitions["WITH_STATIC_LIB"] = not self.options.shared
+        self._cmake.definitions["BOOST_ROOT"] = self.deps_cpp_info['boost'].rootpath
+        self._cmake.definitions["BUILD_TESTING"] = False
+        self._cmake.definitions["BUILD_COMPILER"] = True
+        self._cmake.definitions["BUILD_LIBRARIES"] = True
+        self._cmake.definitions["BUILD_EXAMPLES"] = False
+        self._cmake.definitions["BUILD_TUTORIALS"] = False
+
+        # Make optional libs "findable"
+        if self.options.with_openssl:
+            self._cmake.definitions["OPENSSL_ROOT_DIR"] = self.deps_cpp_info['openssl'].rootpath
+        if self.options.with_zlib:
+            self._cmake.definitions["ZLIB_ROOT"] = self.deps_cpp_info['zlib'].rootpath
+        if self.options.with_libevent:
+            self._cmake.definitions["LIBEVENT_ROOT"] = self.deps_cpp_info['libevent'].rootpath
+
+        self._cmake.configure(build_folder=self._build_subfolder)
+        return self._cmake
+
+    def package(self):
+        self.copy(pattern="LICENSE", dst="licenses", src=self._source_subfolder)
+        cmake = self._configure_cmake()
+        cmake.install()
+        # Copy generated headers from build tree
+        build_source_dir = os.path.join(self._build_subfolder, self._source_subfolder)
+        self.copy(pattern="*.h", dst="include", src=build_source_dir, keep_path=True)
+        tools.rmdir(os.path.join(self.package_folder, "lib", "cmake"))
+        tools.rmdir(os.path.join(self.package_folder, "lib", "pkgconfig"))
+
+    def package_info(self):
+        self.cpp_info.libs = tools.collect_libs(self)
+        # Make sure libs are link in correct order. Important thing is that libthrift/thrift is last
+        # (a little naive to sort, but libthrift/thrift should end up last since rest of the libs extend it with an abbrevation: 'thriftnb', 'thriftz')
+        # The library that needs symbols must be first, then the library that resolves the symbols should come after.
+        self.cpp_info.libs.sort(reverse = True)
+
+        if self.settings.os == "Windows":
+            # To avoid error C2589: '(' : illegal token on right side of '::'
+            self.cpp_info.defines.append("NOMINMAX")
+        elif self.settings.os == "Linux":
+            self.cpp_info.system_libs.extend(["m", "pthread"])
```

Comment:
Why there is not reference for package_folder/bin in PATH ?

Reply:
because virtualrunenv does it anyway?

Reply:
but if I don't use virtualrunenv and I only get the env vars, it won't be there. Please, all env_info.PATH as for any other package: https://github.com/conan-io/conan-center-index/blob/7588836758e9cddf4c97cdf7ca9204ab4bc86cce/recipes/autoconf/all/conanfile.py#L78

Reply:
Do all recipe set LD_LIBRARY_PATH  and DYLD_LIBRARY_PATH ? because it's needed too if you want to be able to run thrift and thrift or any of its dependencies is shared.

Reply:
Reading again the [documentation](https://docs.conan.io/en/latest/reference/build_helpers/run_environment.html), I think we are good here.